service/controller/resource/configmapmigration/create.go

```diff
@@ -95,26 +94,18 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		return microerror.Mask(err)
 	}
 
-	// Get all configmaps in kube-system in the tenant cluster. The migration needs to
-	// complete before we create app CRs. So we cancel the entire loop on error.
-	tenantConfigMaps, err := cc.Client.TenantCluster.K8s.CoreV1().ConfigMaps(metav1.NamespaceSystem).List(listOptions)
-	if tenant.IsAPINotAvailable(err) {
-		r.logger.LogCtx(ctx, "level", "debug", "message", "tenant cluster is not available yet")
-		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
-		return nil
-	} else if err != nil {
-		return microerror.Mask(err)
-	}
-	if errors.Is(ctx.Err(), context.DeadlineExceeded) {
-		r.logger.LogCtx(ctx, "level", "debug", "message", "timeout getting chartconfig CRs")
-		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
-		return nil
-	}
-
 	for _, chartSpec := range chartSpecsToMigrate {
 		if chartSpec.UserConfigMapName != "" {
 			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("finding out if user configmap %#q has been migrated", chartSpec.UserConfigMapName))
 
+			_, err = getChartConfigByName(chartConfigs.Items, chartSpec.ChartName)
+			if IsNotFound(err) {
+				// We delete the chartconfig CR once the migration process is
+				// complete. So if its gone we should not copy the user values
+				// again.
+				r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("user configmap %#q has already been migrated", chartSpec.UserConfigMapName))
+			}
```

Comment:
We delete the chartconfig CR once the migration is complete.

So this is the best way to determine if the migration is done. If it is then we shouldn't continue to copy the config.